File: ChasRooms.Server/Domain/Entities/Room.cs
namespace ChasRooms.Server.Domain.Entities
{
    public class Room
    {
        public int Id { get; set; }
        public string RoomName { get; set; } = null!;
        public int Capacity { get; set; }
        public string features { get; set; } = null!;

        ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    }
}

File: frontend/src/types/room.ts
export type RoomStatus = "available" | "occupied";

export type Room = {
    id: number;
    previewUrl?: string;
    roomName: string;
    capacity: number;
    features: string;
    isOccupied: boolean;
};

export type RoomById = {
    id: number;
    previewUrl?: string;
    name: string;
    capacity: number;
    resources: string[];
    isOccupied: boolean;
    bookings: [
        {
            start: Date;
            end: Date;
        },
    ];
};
